CONTRIBUTING.md

```diff
@@ -0,0 +1,169 @@
+
+![Group 3](https://github.com/user-attachments/assets/7c2b8d15-87cf-4cc7-be86-5fadaebfad0b)
+
+## Step 2: Clone the repository to your local machine
+
+```
+git clone https://github.com/<your-GitHub-username>/CopilotKit
+
+```
+
+![Group 4](https://github.com/user-attachments/assets/e3e78b2b-eead-463b-858b-8d40e4cb18e9)
+
+## Step 3: Prepare the development environment
+
+### 1)Install Prerequisites
+- Node.js 20.x or later
+- pnpm v9.x installed globally (npm i -g pnpm@^9)
+- Turborepo v2.x installed globally (npm i -g turbo@2)
+
+### 2)Install Dependencies
+To install the dependencies using pnpm
+Go inside project folder and run :
+
+```jsx
+pnpm install
+```
+### 3)Build Packages
+To make sure everything works, let’s build all packages once:
+
+```jsx
+turbo run build
+```
+
+## Step 4: Create a branch
+
+Create a new branch for your changes.
+In order to keep branch names uniform and easy-to-understand, please use the following conventions for branch naming.
+Generally speaking, it is a good idea to add a group/type prefix to a branch.
+Here is a list of good examples:
+
+- for docs change : docs/<ISSUE_NUMBER>-<CUSTOM_NAME>
+- for new features : feat/<ISSUE_NUMBER>-<CUSTOM_NAME>
+- for bug fixes : fix/<ISSUE_NUMBER>-<CUSTOM_NAME>
+
+```jsx
+git checkout -b <new-branch-name-here>
+```
+
+## Step 5: Make your changes
+
+Now that everything is set up and works as expected, you can get start developing or update the code with your bug fix or new feature.
+
+```jsx
+# To start all packages in development mode
+turbo run dev
+ 
+# Start a specific package in development mode
+turbo run dev --filter="@copilotkit/package-name"
+```
+
+## Step 6: Add the changes that are ready to be committed
+
+Stage the changes that are ready to be committed:
+
+```jsx
+git add .
+```
+
+## Step 7: Commit the changes (Git)
+
+Commit the changes with a short message. (See below for more details on how we structure our commit messages)
+
+```jsx
+git commit -m "<type>(<package>): <subject>"
+```
+
+## Step 8: Push the changes to the remote repository
+
+Push the changes to the remote repository using:
+
+```jsx
+git push origin <branch-name-here>
+```
+
+## Step 9: Create Pull Request
+
+In GitHub, do the following to submit a pull request to the upstream repository:
+
+1.  Give the pull request a title and a short description of the changes made. Include also the issue or bug number associated with your change. Explain the changes that you made, any issues you think exist with the pull request you made, and any questions you have for the maintainer.
+
+Remember, it's okay if your pull request is not perfect (no pull request ever is). The reviewer will be able to help you fix any problems and improve it!
+
+2.  Wait for the pull request to be reviewed by a maintainer.
+
+3.  Make changes to the pull request if the reviewing maintainer recommends them.
+
+Celebrate your success after your pull request is merged :-)
+
+## Git Commit Messages
```

Comment:
I think you should add this as well into above section inside 'Commit the changes'